### PR TITLE
add compat annotation to NamedTuple macro docs

### PR DESCRIPTION
**Pull request by stevengj**
(Missing from #34548.)

### CODE REVIEW

File: base/namedtuple.jl [hunk 1]
```diff
@@ -350,6 +350,9 @@ julia> @NamedTuple begin
        end
 NamedTuple{(:a, :b),Tuple{Float64,String}}
 ```
+
+!!! compat "Julia 1.5"
```
Comment (JeffBezanson) [inline base/namedtuple.jl]: Trailing whitespace.

Comment (stevengj) [reply base/namedtuple.jl]: whoops, downside of the github in-browser editor

**Comment by KristofferC** [comment]
I think this needs a rebase on master to fix CI

**Comment by stevengj** [comment]
Rebased.

**Pull request merged**